taskfiles/deps/lock.yaml

```diff
@@ -0,0 +1,30 @@
+version: "3"
+
+includes:
+  toolchains: "../toolchains.yaml"
+
+tasks:
+  check-rust:
+    cmds:
+      - task: "cargo-workspace-update"
+        vars:
+          CARGO_UPDATE_FLAGS: "--locked"
+
+  update-rust:
+    cmds:
+      - task: "cargo-workspace-update"
+        vars:
+          CARGO_UPDATE_FLAGS: ""
+
+  # Runs `cargo update` in the root Rust workspace directory with the specified flags.
+  #
+  # @param {string} CARGO_UPDATE_FLAGS The flags to pass to the `cargo update` command.
+  cargo-workspace-update:
+    internal: true
+    requires:
+      vars: ["CARGO_UPDATE_FLAGS"]
+    dir: "{{.ROOT_DIR}}"
+    deps: ["toolchains:rust"]
+    cmd: |-
+      . "$HOME/.cargo/env"
```

Comment:
just curious, do we have plans to install rust into `clp/build/` instead of the user's home directory?